### PR TITLE
fix(metrics): Move metrics calculations back to client

### LINKED ISSUE #60692: Indexer: improve telemetry

### LINKED ISSUE #61537: Indexer: update value length and message size metrics to use weighted average indead of average of average
Currently the indexer metrics for value length and message size works as follows:
For each batch create a mapping of:
- Use case id → value length
- Use case id → message size
- Use case id → number of messages seen

At the end of the batch, we divide the value length / # tag key for a use case with the number of messages for that use case we have seen. This works for tacking the metrics for each individual use case, but it doesn't work for tracking the total average because the actual average needs to be weighted.

### PR DESCRIPTION
**Pull request by john-z-yang**
### Overview

The previous changes made to the metrics that the indexer emit is not as good as I like. It suffers from 2 problems:
1. During deployment the metrics becomes really spiky, which causes the graph to not be very useful when zooming out to see values over the last few days.
2. The average values are broken because we are emitting the sum over a batch

This pr moves the calculation to client side

### CODE REVIEW

**Review by evanh — COMMENTED** [review]
Are there alerts or anything set up that tracks these metrics? Is something going to be unhappy when the old values stop being reported?

File: src/sentry/sentry_metrics/consumers/indexer/batch.py [hunk 1]
```diff
@@ -504,23 +517,54 @@ def reconstruct_messages(
...
             )
             metrics.distribution(
-                "metrics_consumer.process_message.message.size",
-                self.__message_bytes[use_case_id],
+                "metrics_consumer.process_message.message.avg_size_in_batch",
+                self.__message_bytes[use_case_id] / self.__message_count[use_case_id],
                 tags={"use_case_id": use_case_id.value},
                 unit="byte",
             )
             metrics.distribution(
```
Comment (ayirr7) [inline src/sentry/sentry_metrics/consumers/indexer/batch.py:525]: I may be forgetting something, but is there a reason why these have to be a distribution?

**Review by ayirr7 — APPROVED** [review]

File: src/sentry/sentry_metrics/consumers/indexer/batch.py (hunk 1 above)
Comment (john-z-yang) [reply src/sentry/sentry_metrics/consumers/indexer/batch.py:525]: Yes, the second average would be for when the data points are being displayed

**Comment by john-z-yang** [comment]
> Are there alerts or anything set up that tracks these metrics? Is something going to be unhappy when the old values stop being reported?

We have not setup any alerts yet, we will be doing so sometimes today or tomorrow. The current way we are calculating these values is that we send the sum of the values to datadog and then we divide it by the number of of metrics we see. But for some reason, this results in the values being different depending on the "zoom level" of the graph. You can see it in "average message tag counts per batch" graph for example.

**Pull request merged**